src/quacc/runners/ase.py

```diff
@@ -240,6 +246,132 @@ def run_opt(
     return dyn
 
 
+def run_md(
+    atoms: Atoms,
+    timestep: float = 1.0,
+    steps: int = 500,
+    dynamics: MolecularDynamics = VelocityVerlet,
+    dynamics_kwargs: dict[str, Any] | None = None,
+    copy_files: str | Path | list[str | Path] | None = None,
+) -> MolecularDynamics:
+    """
+    Run an ASE-based MD in a scratch directory and copy the results back to
+    the original directory. This can be useful if file I/O is slow in the working
+    directory, so long as file transfer speeds are reasonable.
+
+    This is a wrapper around the dynamical object in ASE. Note: This function does not
+    modify the atoms object in-place.
+
+    Parameters
+    ----------
+    atoms
+        The Atoms object to run the calculation on.
+    timestep
+        The time step in femtoseconds.
+    steps
+        Maximum number of steps to run
+    dynamics
+        MolecularDynamics class to use, from `ase.md.md.MolecularDynamics`.
+    dynamics_kwargs
+        Dictionary of kwargs for the dynamics. Takes all valid kwargs for ASE
+        MolecularDynamics classes.
+    copy_files
+        Filenames to copy from source to scratch directory.
+
+    Returns
+    -------
+    Dymamics
+        The ASE MolecularDynamics object.
+    """
+
+    # Copy atoms so we don't modify it in-place
+    atoms = copy_atoms(atoms)
+
+    # Perform staging operations
+    tmpdir, job_results_dir = calc_setup(atoms, copy_files=copy_files)
+
+    # Set defaults
+    dynamics_kwargs = recursive_dict_merge(
+        {"logfile": "-" if SETTINGS.DEBUG else tmpdir / "dyn.log"}, dynamics_kwargs
+    )
+
+    # Check if trajectory kwarg is specified
+    if "trajectory" in dynamics_kwargs:
+        msg = "Quacc does not support setting the `trajectory` kwarg."
+        raise ValueError(msg)
+
+    if "temperature" in dynamics_kwargs or "temp" in dynamics_kwargs:
+        LOGGER.warning(
+            r"The `temperature`\`temp` kwargs are ASE deprecated and will"
+            "be interpreted as `temperature_K` in Quacc."
+        )
+        dynamics_kwargs["temperature_K"] = dynamics_kwargs.pop(
+            "temperature", None
+        ) or dynamics_kwargs.pop("temp", None)
+
+    if "pressure" in dynamics_kwargs:
+        LOGGER.warning(
```

Comment:
I am happy to do this, but in case you're curious, here is how you write a test for a LOGGER:


```python
LOGGER = logging.getLogger(__name__)
LOGGER.propagate = True

def my_test(caplog):
    with caplog.at_level(logging.WARNING):
        function_that_triggers_warning(arg1, arg2, ...)
```